src/main/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -30,57 +30,63 @@
+    HashMap<Long, String> productsTotalReverse = new HashMap<>();
+    HashMap<String, Long> usersTotal = new HashMap<>();
+    List<String> products = new ArrayList<>();
+    List<String> users = new ArrayList<>();
+    List<String> scores = new ArrayList<>();
     PrintWriter out = new PrintWriter(new FileWriter("movies.csv"));
 
     public MovieRecommender (String filename) throws IOException{
         this.filename = filename;
         getData();
     }
 
-    public void getData() throws IOException{
-        String line;
-        String idProduct="";
-        String idUser="";
-        String score;
-        InputStream fileStream = new FileInputStream(filename);
-        InputStream gzipStream = new GZIPInputStream(fileStream);
-        Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
-        BufferedReader buffered = new BufferedReader(decoder);
+    public void getData(){
+        try{
+            String line;
+            String idProduct="";
+            String idUser="";
+            String score;
+            InputStream fileStream = new FileInputStream(filename);
+            InputStream gzipStream = new GZIPInputStream(fileStream);
+            Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
+            BufferedReader buffered = new BufferedReader(decoder);
 
-        while ((line = buffered.readLine()) != null){
-            if(line.indexOf("product/productId:") != -1){
-                reviews++;
-                idProduct = line.replace("product/productId: ","");
-                if(!productsTotal.containsKey(idProduct)){
-                    productsTotal.put(idProduct, totalProd);
-                    totalProd++;
+            while ((line = buffered.readLine()) != null){
+                if(line.indexOf("product/productId:") != -1){
+                    reviews++;
+                    idProduct = line.replace("product/productId: ","");
+                    if(!productsTotal.containsKey(idProduct)){
+                        productsTotal.put(idProduct, totalProd);
+                        totalProd++;
+                    }
                 }
-            }
-            if(line.indexOf("review/userId:") != -1){
-                idUser = line.replace("review/userId: ","");
-                if(!usersTotal.containsKey(idUser)){
-                    usersTotal.put(idUser, totalUsers);
-                    totalUsers++;
+                if(line.indexOf("review/userId:") != -1){
+                    idUser = line.replace("review/userId: ","");
+                    if(!usersTotal.containsKey(idUser)){
+                        usersTotal.put(idUser, totalUsers);
+                        totalUsers++;
+                    }
                 }
+                if(line.indexOf("review/score") != -1){
+                    score = line.replace("review/score: ","");
+                    scores.add(score);
+                    out.println(usersTotal.get(idUser) + "," +productsTotal.get(idProduct)+ "," + score);
```

Comment:
Also, I think that you will have a lot of lines printed here. Do you think is worthy to have too much messages in log files or console? Which is the true value of logging this info?